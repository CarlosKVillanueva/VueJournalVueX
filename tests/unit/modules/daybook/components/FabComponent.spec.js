import FabComponent from "@/modules/daybook/components/FabComponent";
import { shallowMount } from "@vue/test-utils"


describe( 'Fab Component Unit Test', () => {

    test( 'Must render Snapshot properly', () => {
        const wrapper = shallowMount( FabComponent )

        expect( wrapper.html() ).toMatchSnapshot()
    } );

    test( 'Must show default icon', () => {
        const wrapper = shallowMount( FabComponent )
        const iTag = wrapper.find( 'i' )

        expect( iTag.classes( 'fa-plus' ) ).toBeTruthy()
    } );

    test( 'Must show argumen icon: fa-circle', () => {
        const wrapper = shallowMount( FabComponent, {
            props: {
                icon: 'fa-circle'
            }
        } )
        const iTag = wrapper.find( 'i' )

        expect( iTag.classes( 'fa-circle' ) ).toBeTruthy()
    } );

    test( 'Must emit on:click event when click', () => {
        const wrapper = shallowMount( FabComponent )

        wrapper.find('button').trigger('click')

        // console.log(wrapper.emitted('on:click'))
        expect( wrapper.emitted('on:click') ).toHaveLength(1)
    } );


} );