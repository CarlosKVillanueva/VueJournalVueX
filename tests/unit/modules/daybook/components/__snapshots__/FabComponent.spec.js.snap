// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Component Unit Test Must render Snapshot properly 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
